### ISSUE TITLE
[4.0] Installing languages doesn't work

### ISSUE THREAD

**Issue by Bakual** [joomla/joomla-cms#16337]
### Steps to reproduce the issue
Try installing an additional language during installation or using the Install Language view. It will not find a package.

### Expected result
Install the language

### Actual result
Doesn't install as there is no matching package for 4.0

### Additional comments
Now that is somewhat expected at this point since the targetplatform in http://update.joomla.org/language/translationlist_3.xml doesn't contain 4.0, and thus rightfully no package is found.

As you see from the name of the XML, it is meant for the J3 series. For J4 we may want to use a new XML source. But that means some code changes and changing of update servers in the database.
We could also use the same XML but add a second item for 4.0 for each release.
We likely will have separate language packs for J4.0 that are not compatible with J3.x (due to removed and changed strings).

That needs a decision and setting up of the respective infrastructure (preferrably on JDownloads) so multilingual features can be tested properly on J4. Would be great if that is ready soon.

**Comment by mbabker** [comment]
Without some revisiting of the specs for what we want language package management to look like with the downloads site, and some support in the code work that will take (i.e. if there's a request to use the site frontend, ARS has no frontend management; we also have to extend ARS' ACL down a level otherwise anyone with component access can get anywhere and we don't want that), I don't know if we'll have that site ready to cope with language packages for 4.0.  It'd be nice if we could get it there though.

**Comment by Bakual** [comment]
For reference, the extension we use on our download site is https://github.com/joomla/downloads.joomla.org

**Comment by infograf768** [comment]
> I don't know if we'll have that site ready to cope with language packages for 4.0. It'd be nice if we could get it there though.

In the meanwhile, maybe we could implement a cron that would pick 2 LTR and one RTL language packs from the 3.x joomlacode repo(German, French, Persian) and copy them somewhere where a new xml would fetch them and make them available in 4.0?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16337">issues.joomla.org/tracker/joomla-cms/16337</a>.</sub>

**Comment by mbabker** [comment]
Is this a valid update definition?  I'd rather add some code to make 4.0 allowed on the current list for select languages without setting up more infrastructure just so we can test things.

```xml
  <update>
    <name>German DE</name>
    <description>German DE Translation of Joomla!</description>
    <element>pkg_de-DE</element>
    <type>package</type>
    <version>3.7.2.1</version>
    <downloads>
      <downloadurl type="full" format="zip">http://joomlacode.org/gf/download/frsrelease/20534/163898/de-DE_joomla_lang_full_3.7.2v1.zip</downloadurl>
    </downloads>
    <targetplatform name="joomla" version="3.[0123456789]"/>
    <targetplatform name="joomla" version="4.0"/>
  </update>
```

**Comment by mbabker** [comment]
The other option, assuming the regex is valid, is this:

```xml
    <targetplatform name="joomla" version="[34].[0123456789]"/>
```

**Comment by mbabker** [comment]
Got my answer, last one works.  The cron is updated to make those 3 languages available on 4.0 so when it runs next that (and support for 3.8/9) should be there.

**Comment by Bakual** [comment]
Just for reference, if you wanted to make the package available for 3.x and only 4.0 you would have to duplicate the entry like this:
````
  <update>
    <name>German DE</name>
    <description>German DE Translation of Joomla!</description>
    <element>pkg_de-DE</element>
    <type>package</type>
    <version>3.7.2.1</version>
    <downloads>
      <downloadurl type="full" format="zip">http://joomlacode.org/gf/download/frsrelease/20534/163898/de-DE_joomla_lang_full_3.7.2v1.zip</downloadurl>
    </downloads>
    <targetplatform name="joomla" version="4.0"/>
  </update>
  <update>
    <name>German DE</name>
    <description>German DE Translation of Joomla!</description>
    <element>pkg_de-DE</element>
    <type>package</type>
    <version>3.7.2.1</version>
    <downloads>
      <downloadurl type="full" format="zip">http://joomlacode.org/gf/download/frsrelease/20534/163898/de-DE_joomla_lang_full_3.7.2v1.zip</downloadurl>
    </downloads>
    <targetplatform name="joomla" version="3.[0123456789]"/>
  </update>
````
For testing purposes the solution from Michael of course works fine and is the simplest approach.

**Comment by fouadfana** [comment]
is there an issue with language installer nowadays? I'm using Joomla 3.7.2, as I'm trying to install new language from extension manager but the following warning and error has been shown:

Warning
Error connecting to the server: 404

Error

Invalid URL
Unable to find install package

I tried to download the language package form (https://community.joomla.org/translations/joomla-3-translations.html#ar-aa) to installed offline but it seems all package links don't existing or not reachable.

please any advice regarding to this issue.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16337">issues.joomla.org/tracker/joomla-cms/16337</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
@fouadfana please open a new Issue as this one is for 4.0, thanks.

**Comment by Bakual** [comment]
Works fine for me on 3.7.2. It was either a short outtage from JoomlaCode or something with your connection.

**Comment by fouadfana** [comment]
thanks @frankmayer and @Bakual , it was something with the connection, it seems like restricted to connected with joonle website.

**Comment by mbabker** [comment]
This should actually stay open.  We don't have any of our backend architecture for supporting language packs on 4.0 set up yet (either in the CMS itself or our `joomla.org` web properties).  So this can and should remain the tracking item for that work.

**Comment by franz-wohlkoenig** [comment]
Installed nightly Build full Package using French and Persian fur Multilang: Installation of Lang stopped at Tab 2, couldn't finish Installation of Lang. In Backend Menues for French and Persian are created.

**Comment by brianteeman** [comment]
closed - see #21712 as a dedicated tracker for infrastructure
